### PR TITLE
added tests to hz-cloud

### PR DESCRIPTION
**Pull request by bilalyasar**

### CODE REVIEW

File: hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java [hunk 1]
```diff
@@ -47,6 +50,35 @@
 
 
     @Test
+    public void test_Constasts(){
+        Constants constants = new Constants();
```
Comment (mesutcelik) [inline hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java]: is this a trick to make sonar happy with static fields?

File: hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java [hunk 2]
```diff
@@ -47,6 +50,35 @@
...
+    public void testEC2RequestSigner_whenTimeSignServiceIsNull(){
+        new EC2RequestSigner(new AwsConfig(),"","").sign(null,new HashMap<String, String>());
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void testEC2RequestSigner_whenTimeSignAttributeIsNull(){
+        new EC2RequestSigner(new AwsConfig(),"","").sign("",null);
+    }
+    
+    @Test
```
Comment (mesutcelik) [inline hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java]: can you check `deriveSigningKeyTest` again? It has two big try/catch block that seems unrelated to each other? This test can be simplified.

File: hazelcast-cloud/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java [hunk 3]
```diff
@@ -43,19 +43,6 @@
...
-        }
-        String result = query.toString();
-        if (result != null && !result.equals("")) {
-            result = "?" + result.substring(0, result.length() - 1);
-        }
-        return result;
-    }
-
     public static Map<String, String> unmarshalTheResponse(InputStream stream, AwsConfig awsConfig) throws IOException {
         return parseAddresses(stream, awsConfig);
```
Comment (pveentjer) [inline hazelcast-cloud/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java]: Just copy the content of parseAddresses into the unmarshallTheResponse method.  Since the parseAddresses is not used in any other place and the unmarshalTheResponse is not adding any extra logic, it is best to get rid of complexity.

File: hazelcast-cloud/src/test/java/com/hazelcast/aws/DescribeInstancesTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    @Test(expected = IllegalArgumentException.class)
+    public void testDescribeInstances_whenAwsConfigIsNull() {
+        new DescribeInstances(null,"endpoint");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testDescribeInstances_whenAccessKeyNull() {
+        new DescribeInstances(new AwsConfig(),"endpoint");
+    }
+
```
Comment (pveentjer) [inline hazelcast-cloud/src/test/java/com/hazelcast/aws/DescribeInstancesTest.java]: These are tests for the unhappy situations; which is good to have. But what about the happy situations? Currently we have no idea if that is working.

PS:
Your test is called DescribeInstances; so you are already indicating that you are testing this. That is why you don't need to repeat it in the test name.

So it can be simplified to:
test_whenAwsConfigIsNull

Or if you want to be even shorter:
whenAwsConfigIsNull

Also make sure you do a code reformat.

File: hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java [hunk 5]
```diff
@@ -47,6 +50,35 @@
...
     @Test
+    public void test_Constasts(){
+        Constants constants = new Constants();
+        assertNotNull(constants.DATE_FORMAT);
+        assertNotNull(constants.DOC_VERSION);
+        assertNotNull(constants.SIGNATURE_METHOD_V4);
+        assertNotNull(constants.GET);
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
```
Comment (pveentjer) [inline hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java]: There is no need to repeat EC2RequestSigner in the method names.

File: hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java [hunk 6]
```diff
@@ -47,6 +50,35 @@
...
+        new EC2RequestSigner(null,"","");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testEC2RequestSigner_whenTimeStampIsNull(){
+        new EC2RequestSigner(new AwsConfig(),null,"");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testEC2RequestSigner_whenTimeSignServiceIsNull(){
```
Comment (pveentjer) [inline hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java]: Please don't use such one liners. The following code is a lot easier to read:

```
@Test(expected = IllegalArgumentException.class)
public void testEC2RequestSigner_whenTimeSignServiceIsNull(){
     EC2RequestSigner signer = new EC2RequestSigner(new AwsConfig(),"","");

     signer.sign(null, new HashMap<String, String>());
}
```

File: hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java [hunk 7]
```diff
@@ -47,6 +50,35 @@
...
+        new EC2RequestSigner(new AwsConfig(),"","").sign(null,new HashMap<String, String>());
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void testEC2RequestSigner_whenTimeSignAttributeIsNull(){
+        new EC2RequestSigner(new AwsConfig(),"","").sign("",null);
```
Comment (pveentjer) [inline hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java]: Same here. Shorter code doesn't mean better readable code.

File: hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java (hunk 1 above)
Comment (bilalyasar) [inline hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java]: yes

File: hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java (hunk 5 above)
Comment (bilalyasar) [inline hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java]: thanks, will update.

File: hazelcast-cloud/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java (hunk 3 above)
Comment (bilalyasar) [inline hazelcast-cloud/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java]: logical, will update

File: hazelcast-cloud/src/test/java/com/hazelcast/aws/DescribeInstancesTest.java (hunk 4 above)
Comment (bilalyasar) [inline hazelcast-cloud/src/test/java/com/hazelcast/aws/DescribeInstancesTest.java]: thanks, will update.

File: hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java (hunk 2 above)
Comment (bilalyasar) [inline hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java]: will try to simplify, thanks for the warning.

**Comment by bilalyasar** [comment]
@mesutcelik , @pveentjer updated, thanks for review

**Comment by bilalyasar** [comment]
run-parallel

**Comment by bilalyasar** [comment]
verify

**Comment by mesutcelik** [comment]
verify

**Comment by bilalyasar** [comment]
verify
